internal/armada/server/lease.go

```diff
@@ -162,3 +192,29 @@ func (q *AggregatedQueueServer) ReportDone(ctx context.Context, idList *api.IdLi
 	}
 	return &api.IdList{cleanedIds}, returnedError
 }
+
+func (q *AggregatedQueueServer) reportFailure(jobId string, clusterId string, reason string) error {
+	job, err := getJobById(q.jobRepository, jobId)
+	if err != nil {
+		return err
+	}
+
+	failureReason := reason
+	err = reportFailed(q.eventStore, clusterId, failureReason, job)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getJobById(jobRepository repository.JobRepository, jobId string) (*api.Job, error) {
```

Comment:
maybe make this a method on AggregatedQueueServer or extend the repository to do that.

Reply:
I wouldn't think it worth to extend the JobRepository interface with that method, let me know if you wish me to